dev-docs/best-practices/ui.md

```diff
@@ -0,0 +1,68 @@
+# UI
+
+The Taskcluster UI is implemented in the `ui/` directory of this repository.
+When we have a good reason to not follow the best practices in the UI, we document why.
+
+## Contents
+* [Tables](#tables)
+* [Links](#links)
+* [Buttons](#buttons)
+* [Text](#text)
+* [Spacing](#spacing)
+* [Color](#color)
+* [Visual Hierarchy](#visual-hierarchy)
+* [Document Header Hierarchy](#document-header-hierarchy)
+* [Tracking Changes](#tracking-changes)
+
+## Tables
+* Avoid multi-line text in a table - too much information for the given space. Alternatives:
+  * A [Card](https://material-ui.com/components/cards/).
+  * A [Drawer](https://material-ui.com/components/drawers/) that opens up via a button click in the table.
+* Right-align numbers - it’s easier to scan a column of numbers if the decimal part starts in the same location.
+
+## Links
+* Calculate link URLs before the user clicks on them. By doing so, a user will be able to do things like open links in a
+new tab or have the ability to inspect link URLs prior to the clicking.
+
+## Buttons
+* Destructive actions (e.g., a delete operation) should have a confirmation step.
+It’s an added click but will prevent accidental data loss frustration.
+
+## Text
+* Don’t center text spanning more than 3 lines. It’s not pretty. Alternatives:
+  * Rewrite the content to be shorter.
+  * Left-align the content if the text direction is left-to-right. Otherwise Right-align.
+
+## Spacing
+If you need to add spacing to an element, use the spacing scale defined in the UI as opposed to picking
+a number that looks right to you. This will make sure there is a feel of consistency across the site design.
+
+## Color
+Don’t depend only on color to communicate information, otherwise people who are color blind will have a hard time
+understanding the UI. You can use labels. For things like graphs, you can use different shades of the same color.
+They are easier to differentiate for someone who is color blind since the colors will differ in contrast.
+
+## Visual Hierarchy
+Each button in a view has its own level of importance in that page relative to other actions.
+Most views have a single primary action, a couple that are medium emphasis and some that are rarely used.
+Buttons that are not primary should not have high contrast background colors.
+For example, a destructive button that isn’t considered a primary action in a view destructive does not
```

Comment:
I tend to disagree with this. Red is universally a color of warning. Destructive buttons should not be easy to click on accidentally

Reply:
Another reason why I disagree with this: in the new worker manager UI I have buttons in the table. This is an experiment and something new in our UI. Evidence suggests that those buttons are super-easy to miss if they don't have a background color.

Reply:
I understand your concern.

Our site tries to follow the material design spec. The details for a button can be found [here](https://material.io/design/components/buttons.html#usage). The types of buttons are:

1. Text button (low emphasis)
    - Text buttons are typically used for less important actions.
2. Outlined Button (medium emphasis)
    - Outlined buttons are used for more emphasis than text buttons due to the stroke.
3. Contained button (high emphasis)
    - Contained buttons have more emphasis, as they use use a color fill and shadow.
4. Toggle button
    - Toggle buttons group a set of actions using layout and spacing. They’re used less often than other button types.

>  in the new worker manager UI I have buttons in the table. This is an experiment and something new in our UI. I introduced that to make the UI more convenient.

Sure, I don't think I see a problem with this. I think it's a good thing to have but it's not a primary action for that view. Having the background red will make the icon button a contained one giving it more emphasis than it deserves for that specific view. When a user first enters that view, their eyes will automatically gravitate there first which is not what we want to do. Additionally, I think buttons are hard to miss in a table if they occupy their own column.